elliottlib/cli/find_cve_flaws_cli.py

```diff
@@ -0,0 +1,115 @@
+
+    Also converts advisory to RHSA, if not already.
+
+    Example:
+
+    $ elliott --group openshift-4.6 find-cve-flaws --use-default-advisory image
+    INFO Cloning config data from https://github.com/openshift/ocp-build-data.git
+    INFO Using branch from group.yml: rhaos-4.6-rhel-8
+    INFO found 114 tracker bugs attached to the advisory
+    INFO found 58 corresponding flaw bugs
+    INFO 23 out of 58 flaw bugs considered "first-fix"
+    INFO Adding the following BZs to the advisory: [1880456, 1858827, 1880460,
+    1847310, 1857682, 1857550, 1857551, 1857559, 1848089, 1848092, 1849503,
+    1851422, 1866148, 1858981, 1852331, 1861044, 1857081, 1857977, 1848647,
+    1849044, 1856529, 1843575, 1840253]
+    """
+    runtime.initialize()
+    bzurl = runtime.gitdata.load_data(key='bugzilla').data['server']
+    bzapi = bugzilla.Bugzilla(bzurl)
+
+    if bool(default_advisory_type) == bool(advisory_id):
+        print("Exactly one of '--add' or '--use-default-advisory' options must be specified.")
+        exit(1)
+
+    if default_advisory_type is not None:
+        advisory_id = find_default_advisory(runtime, default_advisory_type)
+
+    attached_tracker_bugs = get_attached_tracker_bugs(bzapi, advisory_id)
+    # # @TODO: remove this block before merge, let it here for now because it
+    # # is useful for testing the PR, when BZs are not yet attached to an advisory
+    # attached_tracker_bugs = bzapi.query(bzapi.build_query(
+    #     product='OpenShift Container Platform',
+    #     status=['MODIFIED', 'ON_QA', 'VERIFIED'],
+    #     target_release='3.11.z',
+    #     keywords=['Security', 'SecurityTracking'],
+    # ))
+    runtime.logger.info('found {} tracker bugs attached to the advisory'.format(
+        len(attached_tracker_bugs)
+    ))
+
+    corresponding_flaw_bugs = get_corresponding_flaw_bugs(bzapi, attached_tracker_bugs)
+    runtime.logger.info('found {} corresponding flaw bugs'.format(
+        len(corresponding_flaw_bugs)
+    ))
+
+    attached_tracker_ids = [tracker.id for tracker in attached_tracker_bugs]
+    current_target_release = "{MAJOR}.{MINOR}.z".format(**runtime.group_config.vars)
+    first_fix_flaw_bugs = [
+        flaw_bug for flaw_bug in corresponding_flaw_bugs
+        if is_first_fix(bzapi, flaw_bug, current_target_release, attached_tracker_ids)
+    ]
+    runtime.logger.info('{} out of {} flaw bugs considered "first-fix"'.format(
+        len(first_fix_flaw_bugs), len(corresponding_flaw_bugs),
+    ))
+
+    if not first_fix_flaw_bugs:
+        runtime.logger.info('No "first-fix" bugs found, exiting')
+        exit(0)
+
+    advisory = Erratum(errata_id=advisory_id)
+    if not is_security_advisory(advisory):
+        runtime.logger.info('Advisory type is {}, converting it to RHSA'.format(advisory.errata_type))
+        cve_boilerplate = runtime.gitdata.load_data(key='erratatool').data['boilerplates']['cve']
+        cves = ' '.join([flaw_bug.alias[0] for flaw_bug in first_fix_flaw_bugs])
```

Comment:
I think the CVEs and security impact should be updated even if it's already an RHSA. We could have a scenario where we ran this then had to update it to add more bugs before releasing. As written this would not be updated.

Reply:
hm, but it is changing the impact in any case ... the code that handles impact is outside of this `if`.

Reply:
security impact is already being handled outside this `if` block, but now I moved the CVEs out, so it would work against advisories that are _already_ RHSAs: https://github.com/openshift/elliott/pull/158/commits/0519d417bfa3757517db25a4f13f2a3340448617